### PR TITLE
Feature/Login Page

### PR DESCRIPTION
**Pull request by sasalx**
Implemented: 

-  Login Page
- Firebase Auth (Login/Logout)
- Inner Logic for Routing (Streaming)

To-Do: 

- Enhance Login Page design.

---

### CODE REVIEW

**Comment by kondanta** [comment]
test if its broken

**Review by kondanta — APPROVED** [review]
Overall it looks nice. But it looks like you're passing down auth like 3 or 4 siblings below. I think using BLoC for that would be more efficient. Still, I'll accept this PR because it's acceptable and usable.

File: pubspec.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ dependencies:
   flip_card: ^0.2.1
   rxdart: ^0.21.0
   flutter_graphql: ^1.0.0-rc.3
+  firebase_auth: ^0.7.0
```
[the remark this answers is not in the document]
Comment (sasalx) [reply pubspec.yaml:15]: Sacrifices were made. However, I applaud your choice of gif.

File: lib/pages/patient/voip.dart [hunk 2]
```diff
@@ -9,7 +9,7 @@ class VoipConnection extends StatefulWidget {
   static String tag = 'call';
 
   // Needs to be set in here.
-  final String ip = '192.168.1.108';
+  final String ip = '192.168.160.73';
```
[the remark this answers is not in the document]
Comment (sasalx) [reply lib/pages/patient/voip.dart:12]: Blame yourself.

File: pubspec.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kondanta) [reply pubspec.yaml:15]: I choose what fits the context.

**Pull request merged**